src/api/providers/openai.ts

```diff
@@ -17,7 +17,8 @@
 	constructor(options: ApiHandlerOptions) {
 		this.options = options
 		// Azure API shape slightly differs from the core API shape: https://github.com/openai/openai-node?tab=readme-ov-file#microsoft-azure-openai
-		if (this.options.openAiBaseUrl?.toLowerCase().includes("azure.com")) {
+		const urlHost = new URL(this.options.openAiBaseUrl).host;
```

Comment:
@ellipsis-dev — Copilot Autofix generated that line.

What would you propose concretely with a suggestion of how that line should be re-written?

Reply:
Let's try this:

Change:
`const urlHost = new URL(this.options.openAiBaseUrl).host;`

To:
`const urlHost = new URL(this.options.openAiBaseUrl ?? "").host;`

because the nullish coalescing operator (??) to default to an empty string if this.options.openAiBaseUrl is undefined.

It ensures that the URL constructor always receives a valid string.

```suggestion
		const urlHost = new URL(this.options.openAiBaseUrl ?? "").host;
```